docs/pages/setup/guides/terraform-provider.mdx

```diff
@@ -0,0 +1,236 @@
+  cluster.
+
+## Prerequisites
+
+- [Terraform >= (=terraform.version=)+](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+- Teleport (=teleport.version=) Cloud, Open Source or Enterprise
+
+```bash
+terraform version
+Terraform v(=terraform.version=)
+```
+
+```bash
+tctl version
+Teleport v(=teleport.version=) go(teleport.golang)
+```
+
+Create a folder `teleport-terraform` to hold some temporary files:
+
+```bash
+mkdir -p teleport-terraform
+```
+
+## Step 1/4. Install terraform provider
+
+<Tabs>
+  <TabItem label="Linux">
+  ```bash
+  mkdir -p ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/linux_amd64
+  curl -L -O https://get.gravitational.com/terraform-provider-teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz
+  tar -zxvf terraform-provider-teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz -C ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/linux_amd64
+  ```
+  </TabItem>
+
+  <TabItem label="MacOS">
+  ```bash
+  mkdir -p ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/darwin_amd64
+  ```
+  </TabItem>
+</Tabs>
+  
+# Step 2/4. Create a terraform user
+
+Put the following content into `terraform.yaml`:
+
+```
+kind: role
+metadata:
+  name: terraform
+spec:
+  allow:
+    rules:
+      - resources: ['user', 'role', 'token', 'trusted_cluster', 'github', 'oidc', 'saml']
+        verbs: ['list','create','read','update','delete']
+version: v3
+---
+kind: user
+metadata:
+  name: terraform
+spec:
+  roles: ['terraform']
+version: v2
+```
+
+Run:
+
+```bash
+tctl create terraform.yaml
+tctl auth sign --format=file --user=terraform --out=terraform-identity --ttl=10h
+```
+
+<Admonition type="note">
+Clients missing impersonation privileges when trying to use `tctl auth sign`,
+will get the following error:
+
+```bash
+ERROR: access denied: impersonation is not allowed
+```
+
+Create the following file with role: `terraform-impersonator.yaml`:
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: terraform-impersonator
+spec:
+  # SSH options used for user sessions 
+  options:
+    # max_session_ttl defines the TTL (time to live) of SSH certificates 
+    # issued to the users with this role.
+    max_session_ttl: 10h
+
+  # allow section declares a list of resource/verb combinations that are
+  # allowed for the users of this role. by default nothing is allowed.
+  allow:
+    impersonate:
+      users: ['terraform']
+      roles: ['terraform']
+
+  # the deny section uses the identical format as the 'allow' section.
+  # the deny rules always override allow rules.
+  deny:
+    node_labels:
+      '*': '*'
+```
+
+```bash
+tctl create terraform-impersonator.yaml
+```
+
+Assign this role to the current user. Re-login to assume the new role and try
+to issue certificate for terraform user again.
+
+</Admonition>
+
+## Step 3/4. Create Terraform configuration
+
+Create a `main.tf` terraform file:
+
+```
```

Comment:
Less a concern than a question about convention. I think we've been putting config files in here: https://github.com/gravitational/teleport/tree/master/examples perhaps some of the configuration should be moved? 

Also, this might be silly, is the intent to direct readers to the new Setup Section then have Guides and/or Getting Started articles refer to them? Or will each of the other articles share some setup content?

Reply:
Each product section will have some product-specific setup content. The setup section will reference setup that is common to all products. There will be some content duplication, we will remediate using includes.